app/models/foreman_openscap/arf_report.rb

```diff
@@ -110,46 +241,34 @@ def self.create_arf(asset, proxy, params)
       return unless policy
 
       ArfReport.transaction do
+        # construct body
+        body = []
+        if params[:logs]
+          params[:logs].each do |log|
+            rule_id, needs_update = Rule.where(label: log[:source]).pluck(:id, :needs_update)&.first
+            if rule_id.nil? || needs_update
+              new_rule = Rule.create!(label: log[:source],
+                title: log[:title],
+                severity: log[:severity],
+                description: log[:description],
+                references: log[:references],
+                rationale: log[:rationale],
+                fixes: log[:fixes],
+                needs_update: false)
+              rule_id = new_rule.id
+            end
+            body << [log[:source], log[:result], rule_id]
+          end
+        end
+        # create report
         arf_report = ArfReport.create(:host => asset.host,
                                       :reported_at => Time.at(params[:date].to_i),
                                       :status => params[:metrics],
                                       :metrics => params[:metrics],
-                                      :openscap_proxy => proxy)
+                                      :openscap_proxy => proxy,
+                                      :body => body.to_json)
         return arf_report unless arf_report.persisted?
         PolicyArfReport.where(:arf_report_id => arf_report.id, :policy_id => policy.id, :digest => params[:digest]).first_or_create!
```

Comment:
Would it make sense to have report_id and digest saved directly in ArfReport since we own the table now and discard PolicyArfReport?